### ISSUE TITLE
[SoftmaxRegression][JoxFix] The output of loss function was not good

### ISSUE THREAD

**Issue by alaricbat** [alaricbat/SoftmaxRegression#1]
The output of loss function was not good.
Please check performance of currently integrated SoftmaxRegression algorithm.
Report day by day for the fixing in below.

**Comment by alaricbat** [comment]
[2025/07/08][AlaricBat]: This thread is created.

**Comment by alaricbat** [comment]
[2025/07/08][AlaricBat]
<img width="285" height="369" alt="Image" src="https://github.com/user-attachments/assets/bfc13e2e-01a7-4c45-8419-7a3fbd5537ab" />
Firstly, every reporting of loss was too high. Please check the problem.

**Comment by alaricbat** [comment]
[2025/07/29][AlaricBat]
Update encoding mechanism of X dataset see below:

<img width="1192" height="495" alt="Image" src="https://github.com/user-attachments/assets/fdbe24c7-b0ce-43c1-bd7a-ce3b491d9a14" />

**Step 1:** Parsing 'AVG', 'POOR', 'BEST' into '1', '2', '3'
**Step 2:** Scaling X data set with [MIN-MAX] algorithm.
**Step 3:** Checking the correction of softmax function, regression.
**Step 4:** Training and testing of trained dataset.

Result is below:

<img width="463" height="198" alt="Image" src="https://github.com/user-attachments/assets/6ff9121e-7d01-458d-a216-4390839380f0" />

=> Continue to check with inputing any single data.

**Comment by alaricbat** [comment]
[2025/08/03][AlaricBat]
Applied with **Min-Max Scaling** was not possible.
It would create translated record like **[0 0 0]** (see below): 

<img width="170" height="36" alt="Image" src="https://github.com/user-attachments/assets/4cde9b73-9393-46fc-b4cb-d78c5c876aaf" />

If it is actually occurring, the estimation would be led to below:

<img width="597" height="281" alt="Image" src="https://github.com/user-attachments/assets/642d1733-f3f5-49fa-ac05-6d248f92ce5b" />

So, it will be try again with other approaches.

**Comment by alaricbat** [comment]
**[2025/08/03][AlaricBat]**

Another reporting from the problem of applying **Min-Max Scaling**:
After encoding with **POOR**, **AVG**, **BEST** labels, it led to marked **1** by always and **IT WAS ROOT CAUSE** for why it should be applied other scaling method.

<img width="740" height="438" alt="Image" src="https://github.com/user-attachments/assets/527c676c-79ea-4d85-bcf3-97a96b04b172" />

**Comment by alaricbat** [comment]
**[2025/08/03][AlaricBat]**

After trying to apply **Standard Deviation** that it also had problem like **Min Max Scaling**:

<img width="733" height="490" alt="Image" src="https://github.com/user-attachments/assets/b85a8725-75ca-4ce2-a608-5fa4c132d5a9" />

So, please using "OneHotCoding" for X.

**Comment by alaricbat** [comment]
**[2025/08/04][AlaricBat]**

After applying one hot encoding => the accuracy reached at **100%**.

<img width="463" height="198" alt="Image" src="https://github.com/user-attachments/assets/b940a06a-22fc-4738-934b-0443a635d747" />

However, cost still be large was at **586.9483466710376**

<img width="736" height="343" alt="Image" src="https://github.com/user-attachments/assets/e638354c-271d-4e57-85dd-6e93eae9dfc2" />

=> It will be created with new ticket.

This, ticket will be close for today report. Please refer new commit: https://github.com/alaricbat/SoftmaxRegression/commit/5064c10a48b26f89e52c8a290978c46792005709
